statesync/syncer.go

```diff
@@ -241,12 +241,15 @@ func (s *syncer) Sync(snapshot *snapshot, chunks *chunkQueue) (sm.State, *types.
 		go s.fetchChunks(ctx, snapshot, chunks)
 	}
 
+	pctx, pcancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer pcancel()
+
```

Comment:
I think you can just use the context from above, with a timeout. If the state provider fetches time out, then there's no point in letting the chunk fetchers continue since the whole operation will error anyway.